src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -152,6 +153,13 @@ private void setParameterMap(
               BUFFER_FLUSH_INTERVAL_IN_MILLIS, MAX_CLIENT_LAG));
     }
 
+    this.checkAndUpdate(
```

Comment:
Add comment that this should be the first thing to be set as the final value of this setting can drive the defaulting/logic further on in this method.

Reply:
Added.

Reply:
Added comment.